java/pom.xml

```diff
@@ -905,6 +905,197 @@
+                  <workingDirectory>../</workingDirectory>
+                </configuration>
+              </execution>
+              <execution>
+                <id>datasetarrowdependency</id>
+                <phase>generate-resources</phase>
+                <goals>
+                  <goal>exec</goal>
+                </goals>
+                <configuration>
+                  <executable>cmake</executable>
+                  <commandlineArgs>
+                    -S cpp
+                    -B cpp-jni
+                    -DARROW_DEPENDENCY_USE_SHARED=OFF
+                    -DARROW_CSV=ON
+                    -DARROW_DATASET=ON
+                    -DARROW_FILESYSTEM=ON
+                    -DARROW_GANDIVA=ON
+                    -DARROW_GANDIVA_JAVA=ON
+                    -DARROW_GANDIVA_STATIC_LIBSTDCPP=ON
+                    -DARROW_JNI=ON
+                    -DARROW_ORC=ON
+                    -DARROW_PARQUET=ON
+                    -DARROW_PLASMA=ON
+                    -DARROW_PLASMA_JAVA_CLIENT=ON
+                    -DARROW_S3=ON
+                    -DARROW_USE_CCACHE=ON
+                    -DCMAKE_BUILD_TYPE=Release
+                    -DCMAKE_INSTALL_LIBDIR=lib
+                    -DCMAKE_INSTALL_PREFIX=java-dist
+                    -DCMAKE_UNITY_BUILD=ON
+                    -DARROW_DEPENDENCY_SOURCE=BUNDLED
+                  </commandlineArgs>
+                  <workingDirectory>../</workingDirectory>
+                  <skip>${cpp.dependencies.builded}</skip>
+                </configuration>
+              </execution>
+              <execution>
+                <id>datasetarrowdependencybuild</id>
+                <phase>generate-resources</phase>
+                <goals>
+                  <goal>exec</goal>
+                </goals>
+                <configuration>
+                  <executable>cmake</executable>
+                  <commandlineArgs>
+                    --build cpp-jni --target install --config Release
+                  </commandlineArgs>
+                  <workingDirectory>../</workingDirectory>
+                  <skip>${cpp.dependencies.builded}</skip>
+                </configuration>
+              </execution>
+              <execution>
+                <id>datasetdefine</id>
+                <phase>generate-resources</phase>
+                <goals>
+                  <goal>exec</goal>
+                </goals>
+                <configuration>
+                  <executable>cmake</executable>
+                  <commandlineArgs>
+                    -S java
+                    -B java-jni
+                    -DCMAKE_BUILD_TYPE=Release
+                    -DCMAKE_INSTALL_PREFIX=${arrow.c.jni.dist.dir}
+                    -DCMAKE_PREFIX_PATH=${project.basedir}/../java-dist
+                    -DARROW_JAVA_JNI_ENABLE_DEFAULT=OFF
+                    -DARROW_JAVA_JNI_ENABLE_DATASET=ON
+                    -DBUILD_TESTING=OFF
```

Comment:
Could you sort this list in alphabetical order?

Reply:
Sorted